File: frontend/types/questionnaire.ts
export type SymptomSeverity = 0 | 1 | 2 | 3

export interface EnergySymptoms {
  fatigue: SymptomSeverity
  morning_grogginess: SymptomSeverity
  afternoon_slump: SymptomSeverity
  cold_limbs: SymptomSeverity
  spontaneous_sweating: SymptomSeverity
  night_sweats: SymptomSeverity
}

export interface DigestionSymptoms {
  bloating: SymptomSeverity
  loose_stools: SymptomSeverity
  constipation: SymptomSeverity
  poor_appetite: SymptomSeverity
  heartburn: SymptomSeverity
  nausea_after_eating: SymptomSeverity
}

export interface MoodSymptoms {
  anxiety: SymptomSeverity
  irritability: SymptomSeverity
  depression: SymptomSeverity
  mental_fog: SymptomSeverity
  sighing: SymptomSeverity
}

export interface PainSymptoms {
  headache: SymptomSeverity
  chest_tightness: SymptomSeverity
  joint_pain: SymptomSeverity
  muscle_aches: SymptomSeverity
  fixed_pain_location: SymptomSeverity
}

export interface SleepSymptoms {
  difficulty_falling_asleep: SymptomSeverity
  frequent_waking: SymptomSeverity
  dream_disturbed: SymptomSeverity
  early_morning_waking: SymptomSeverity
}

export interface QuestionnaireResponse {
  session_id: string
  submitted_at?: string
  energy: EnergySymptoms
  digestion: DigestionSymptoms
  mood: MoodSymptoms
  pain: PainSymptoms
  sleep: SleepSymptoms
  additional_notes: string
}

export const EMPTY_QUESTIONNAIRE: Omit<QuestionnaireResponse, 'session_id'> = {
  energy: { fatigue: 0, morning_grogginess: 0, afternoon_slump: 0, cold_limbs: 0, spontaneous_sweating: 0, night_sweats: 0 },
  digestion: { bloating: 0, loose_stools: 0, constipation: 0, poor_appetite: 0, heartburn: 0, nausea_after_eating: 0 },
  mood: { anxiety: 0, irritability: 0, depression: 0, mental_fog: 0, sighing: 0 },
  pain: { headache: 0, chest_tightness: 0, joint_pain: 0, muscle_aches: 0, fixed_pain_location: 0 },
  sleep: { difficulty_falling_asleep: 0, frequent_waking: 0, dream_disturbed: 0, early_morning_waking: 0 },
  additional_notes: '',
}

File: backend/models/questionnaire.py
from __future__ import annotations
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Literal

SymptomSeverity = Literal[0, 1, 2, 3]


class EnergySymptoms(BaseModel):
    fatigue: SymptomSeverity = 0
    morning_grogginess: SymptomSeverity = 0
    afternoon_slump: SymptomSeverity = 0
    cold_limbs: SymptomSeverity = 0
    spontaneous_sweating: SymptomSeverity = 0
    night_sweats: SymptomSeverity = 0


class DigestionSymptoms(BaseModel):
    bloating: SymptomSeverity = 0
    loose_stools: SymptomSeverity = 0
    constipation: SymptomSeverity = 0
    poor_appetite: SymptomSeverity = 0
    heartburn: SymptomSeverity = 0
    nausea_after_eating: SymptomSeverity = 0


class MoodSymptoms(BaseModel):
    anxiety: SymptomSeverity = 0
    irritability: SymptomSeverity = 0
    depression: SymptomSeverity = 0
    mental_fog: SymptomSeverity = 0
    sighing: SymptomSeverity = 0


class PainSymptoms(BaseModel):
    headache: SymptomSeverity = 0
    chest_tightness: SymptomSeverity = 0
    joint_pain: SymptomSeverity = 0
    muscle_aches: SymptomSeverity = 0
    fixed_pain_location: SymptomSeverity = 0


class SleepSymptoms(BaseModel):
    difficulty_falling_asleep: SymptomSeverity = 0
    frequent_waking: SymptomSeverity = 0
    dream_disturbed: SymptomSeverity = 0
    early_morning_waking: SymptomSeverity = 0


class QuestionnaireResponse(BaseModel):
    session_id: str
    submitted_at: str = Field(default_factory=lambda: datetime.utcnow().isoformat())
    energy: EnergySymptoms = Field(default_factory=EnergySymptoms)
    digestion: DigestionSymptoms = Field(default_factory=DigestionSymptoms)
    mood: MoodSymptoms = Field(default_factory=MoodSymptoms)
    pain: PainSymptoms = Field(default_factory=PainSymptoms)
    sleep: SleepSymptoms = Field(default_factory=SleepSymptoms)
    additional_notes: str = ""
    # Optional ZYYXH/T157-2009 王琦 60-item scale answers (item_id → 1–5).
    # When present, the formal transformed-score scorer overrides symptom-based scoring.
    wangqi_answers: dict[str, int] = Field(default_factory=dict)
